stm/stm.ml

```diff
@@ -2729,12 +2729,29 @@ let rec join_admitted_proofs id =
        join_admitted_proofs view.next
   | _ -> join_admitted_proofs view.next
 
+(* Error resiliency may have tolerated some broken commands *)
+let rec check_no_err_states ~doc visited id =
+  let open Stateid in
+  if Set.mem id visited then visited else
+  match state_of_id ~doc id with
+  | `Error e -> raise e
```

Comment:
@maximedenes heads up since IIRC in your PR about the cache we had a discussion about the representation of erroneous states and here I call the API you have changed.